### PR TITLE
Add support for offset in Explodes

### LINKED ISSUE #18760: Offset support for explodes.
For all those scenarios where where main entity is not on the center of the cell.

### PR DESCRIPTION
**Pull request by abc013**
Closes #18760.

Testcase: Build a TS nod stealth generator and destroy it.

### CODE REVIEW

**Review by abcdefg30 — DISMISSED** [review]

File: OpenRA.Mods.Common/Traits/Explodes.cs [hunk 1]
```diff
@@ -52,6 +52,9 @@ public class ExplodesInfo : ConditionalTraitInfo, Requires<IHealthInfo>
 			"Footprint (explosion on each occupied cell).")]
 		public readonly ExplosionType Type = ExplosionType.CenterPosition;
 
+		[Desc("Offset of the explosion.")]
```
Comment (phrohdoh) [inline OpenRA.Mods.Common/Traits/Explodes.cs]:
```suggestion
		[Desc("Offset of the explosion from the center of the exploding actor (or cell).")]
```

Comment (abcdefg30) [reply OpenRA.Mods.Common/Traits/Explodes.cs]: That isn't entirely correct for footprint based explosions.

Comment (phrohdoh) [reply OpenRA.Mods.Common/Traits/Explodes.cs]:
```
weapon.Impact(Target.FromPos(self.World.Map.CenterOfCell(c) + Info.Offset), source);
```

```
weapon.Impact(Target.FromPos(self.CenterPosition + Info.Offset), source);
```

seems correct to me

Comment (abcdefg30) [reply OpenRA.Mods.Common/Traits/Explodes.cs]: Well, I guess the edit is fine.

Comment (abc013) [reply OpenRA.Mods.Common/Traits/Explodes.cs]: Updated.

File: OpenRA.Mods.Common/Traits/Explodes.cs [hunk 2]
```diff
@@ -52,6 +52,9 @@ public class ExplodesInfo : ConditionalTraitInfo, Requires<IHealthInfo>
 			"Footprint (explosion on each occupied cell).")]
 		public readonly ExplosionType Type = ExplosionType.CenterPosition;
 
+		[Desc("Offset of the explosion from the center of the exploding actor (or cell)..")]
```
Comment (phrohdoh) [inline OpenRA.Mods.Common/Traits/Explodes.cs]: nit
```suggestion
		[Desc("Offset of the explosion from the center of the exploding actor (or cell)")]
```

**Review by phrohdoh — DISMISSED** [review]

File: OpenRA.Mods.Common/Traits/Explodes.cs (hunk 2 above)
Comment (abc013) [reply OpenRA.Mods.Common/Traits/Explodes.cs]: Argh! Sorry, fixed.

**Review by phrohdoh — DISMISSED** [review]

**Comment by abc013** [comment]
Removed testcase.

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/97bd7f67259064bd3c0c08f151494770971a9973...3de2e22c6c7c706d5e4d3b99c1f3267c2d456edf)

**Pull request merged**